### PR TITLE
Update client-failover.md

### PR DESCRIPTION
**Pull request by zryska**
Reflected updates to the client failover facet:
- None is no longer supported failover mode
- Updated properties descriptions
- Removed Id property and edit the wording to not treat it as a collection
- Added PATCH support

### CODE REVIEW

File: shared-content/configuration/client-failover.md [hunk 1]
```diff
@@ -41,19 +43,20 @@ On successful execution, the client failover change takes effect immediately dur
...
-| **TokenEndpoint** | Optional for OCS endpoint | `string`| Retrieves an OCS token from an alternative endpoint <br><br>Allowed value: well-formed http or https endpoint string <br>Default value: `null` |
-| **ValidateEndpointCertificate** | Optional | `bool`| Disables verification of destination certificate.<br><br>**Note:** Only use for testing with self-signed certificates. <br><br>Allowed value: `true` or `false`<br>Default value: `true` |
+| **FailoverGroupId** | Required | `string` | The ID of the failover group to register the adapter instance in <br><br>Allowed value: any string identifier<br>Default value: `null` |
+| **Name** | Optional | `string` | The friendly name of the failover group <br><br>Allowed value: any string value<br>Default value: `null` |
+| **Description** | Optional | `string`| The description of the failover group <br><br>Allowed value: any string value<br>Default value: `null` |
+| **FailoverTimeout** | Required | `datetime` | The failover timeout value of the failover group. This defines how frequently the adapter will send a heartbeat to the failover endpoint. <br><br>Allowed value: a string representation of date time using `hh:mm:ss` <br>|
```
Comment (Xiaoli-OSI) [inline shared-content/configuration/client-failover.md:49]: I recall the heartbeat is sent to the failover endpoint every half of the FailoverTimeout, can we be explicit here?

File: shared-content/configuration/client-failover.md [hunk 2]
```diff
@@ -41,19 +43,20 @@ On successful execution, the client failover change takes effect immediately dur
...
+| **Mode** | Required | `string` | The failover mode of the registered adapter. <br><br>Allowed value: `Hot`, `Warm`, `Cold` <br> For more information, see [Failover Modes](#failover-modes). |
+| **Endpoint** | Required | `string` | Destination that support client failover registration. Supported destinations include ADH and on-premise Failover Service<br><br>Allowed value: well-formed http or https endpoint string<br>Default: `null` |
```
Comment (Xiaoli-OSI) [inline shared-content/configuration/client-failover.md:51]: supports?

File: shared-content/configuration/client-failover.md [hunk 3]
```diff
@@ -41,19 +43,20 @@ On successful execution, the client failover change takes effect immediately dur
...
+| **UserName** | Optional | `string` | Basic authentication to on-premise Failover Service endpoint <br><br>Allowed value: any string<br>Default: `null`<br><br>**Note:** If your username contains a backslash, you must add an escape character, for example, type `OilCompany\TestUser` as `OilCompany\\TestUser`. |
+| **Password** | Optional | `string` | Basic authentication to on-premise Failover Service endpoint <br><br>Allowed value: any string or `{{<secretId>}}` (see [Reference Secrets](xref:ReferenceSecrets))<br>Default: `null` |
```
Comment (Xiaoli-OSI) [inline shared-content/configuration/client-failover.md:53]: Same as above, maybe "Password used for basic authentication to.."?

File: shared-content/configuration/client-failover.md [hunk 4]
```diff
@@ -41,19 +43,20 @@ On successful execution, the client failover change takes effect immediately dur
...
+| **ClientId** | Required for ADH endpoint | `string` | Authentication with the ADH endpoint <br><br>Allowed value: any string, can be null if the endpoint URL schema is `HTTP`<br>Default: `null` |
+| **ClientSecret** | Required for ADH endpoint | `string`| Authentication with the ADH endpoint <br><br>Allowed value: any string or `{{<secretId>}}` (see [Reference Secrets](xref:ReferenceSecrets))<br>Default: `null` |
+| **TokenEndpoint** | Optional | `string`| Retrieves a bearer token from an alternative endpoint <br><br>Allowed value: well-formed http or https endpoint string <br>Default value: `null` |
```
Comment (Xiaoli-OSI) [inline shared-content/configuration/client-failover.md:56]: I would suggest being consistent with the rest of the description. Maybe "An optional token endpoint where an adapter retrieves a bearer token from. When null or not specified, the token endpoint is...?

File: shared-content/configuration/client-failover.md [hunk 5]
```diff
@@ -41,19 +43,20 @@ On successful execution, the client failover change takes effect immediately dur
...
+| **ValidateEndpointCertificate** | Optional | `bool`| Disables verification of the server certificate.<br><br>**Note:** Only use for testing with self-signed certificates. <br><br>Allowed value: `true` or `false`<br>Default value: `true` |
```
Comment (Xiaoli-OSI) [inline shared-content/configuration/client-failover.md:57]: Maybe "An optional Boolean, when set to false, adapter would disable the verification of the server certificate. "?

File: shared-content/configuration/client-failover.md [hunk 6]
```diff
@@ -81,13 +83,11 @@ The following is an example of a complete client failover configuration.
    "Endpoint": "http://test-endpoint.com",
    "UserName": "UserName1",
    "Password": "Password1",
-   "TokenEndpoint": "http://token.com",
+   "TokenEndpoint": null,
    "ValidateEndpointCertificate": false
```
Comment (Xiaoli-OSI) [inline shared-content/configuration/client-failover.md:87]: can we have it set to true in the example?

File: shared-content/configuration/client-failover.md (hunk 3 above)
Comment (zryska) [reply shared-content/configuration/client-failover.md:53]: Yeah that was my thinking too but in the end decided to be consistent with our existing description: https://github.com/osisoft/PI-Adapter/blob/v1.7/shared-content/configuration/egress-endpoints.md. It might be good to change it on all places.

File: shared-content/configuration/client-failover.md [hunk 7]
```diff
@@ -41,19 +43,20 @@ On successful execution, the client failover change takes effect immediately dur
...
+| **Description** | Optional | `string`| The description of the failover group <br><br>Allowed value: any string value<br>Default value: `null` |
+| **FailoverTimeout** | Required | `datetime` | The failover timeout value of the failover group. This defines how frequently the adapter will send a heartbeat to the failover endpoint. <br><br>Allowed value: a string representation of date time using `hh:mm:ss` <br>|
+| **Mode** | Required | `string` | The failover mode of the registered adapter. <br><br>Allowed value: `Hot`, `Warm`, `Cold` <br> For more information, see [Failover Modes](#failover-modes). |
+| **Endpoint** | Required | `string` | Destination that support client failover registration. Supported destinations include ADH and on-premise Failover Service<br><br>Allowed value: well-formed http or https endpoint string<br>Default: `null` |
+| **UserName** | Optional | `string` | Basic authentication to on-premise Failover Service endpoint <br><br>Allowed value: any string<br>Default: `null`<br><br>**Note:** If your username contains a backslash, you must add an escape character, for example, type `OilCompany\TestUser` as `OilCompany\\TestUser`. |
```
[the remark this answers is not in the document]
Comment (zryska) [reply shared-content/configuration/client-failover.md:52]: Yeah that was my thinking too but in the end decided to be consistent with our existing description: https://github.com/osisoft/PI-Adapter/blob/v1.7/shared-content/configuration/egress-endpoints.md. It might be good to change it on all places.

File: shared-content/configuration/client-failover.md (hunk 6 above)
Comment (zryska) [reply shared-content/configuration/client-failover.md:87]: Good catch! I'll change it to true.

File: shared-content/configuration/client-failover.md [hunk 8]
```diff
@@ -13,6 +13,8 @@ Using client failover, you can do the following:
 - Perform runtime failover parameter changes such as the failover mode and failover timeout.
 - Query the current failover state including the failover role, last data process time, failover status and adapter state.
 
+**Note:** Failover group will be created by the adapter in case it does not exist on the failover endpoint.
```
Comment (aveva-jhwang) [inline shared-content/configuration/client-failover.md:16]: Maybe reword it to: "Failover group will be created by the adapter if it does not exist on the failover endpoint."
